source/frontend/src/components/maps/MapLeafletView.tsx

```diff
@@ -1,130 +1,141 @@
-import { MapStateActionTypes, MapStateContext } from './providers/MapStateContext';
 
 export type MapLeafletViewProps = {
-  lat: number;
-  lng: number;
-  zoom?: number;
-  whenCreated?: (map: LeafletMap) => void;
-  whenReady?: () => void;
-  geoFilter?: IGeoSearchParams;
-  mapRef: React.MutableRefObject<LeafletMap | null>;
+  parentWidth: number | undefined;
 };
 
 type BaseLayerFile = {
   basemaps: BaseLayer[];
 };
 
-const defaultBounds = new LatLngBounds([60.09114547, -119.49609429], [48.78370426, -139.35937554]);
-
 /**
  * Creates a Leaflet map and by default includes a number of preconfigured layers.
  * @param param0
  */
 const MapLeafletView: React.FC<React.PropsWithChildren<MapLeafletViewProps>> = ({
-  lat,
-  lng,
-  zoom: zoomProp,
-  whenReady,
-  whenCreated,
-  geoFilter,
-  mapRef,
+  parentWidth,
 }) => {
-  const dispatch = useDispatch();
-
   const [baseLayers, setBaseLayers] = useState<BaseLayer[]>([]);
 
   const [activeBasemap, setActiveBasemap] = useState<BaseLayer | null>(null);
-  const smallScreen = useMediaQuery({ maxWidth: 1800 });
 
   const [bounds, setBounds] = useState<LatLngBounds>(defaultBounds);
-  const [layerPopup, setLayerPopup] = useState<LayerPopupInformation>();
 
   // a reference to the layer popup
   const popupRef = useRef<LeafletPopup>(null);
 
-  const history = useHistory();
-  const onPropertyViewClicked = (pid?: string | null, id?: number) => {
-    if (id !== undefined) {
-      history.push(`/mapview/sidebar/property/${id}?pid=${pid}`);
-    } else if (pid !== undefined && pid !== null) {
-      const parsedPid = pidParser(pid);
-      history.push(`/mapview/sidebar/non-inventory-property/${parsedPid}`);
-    } else {
-      console.warn('Invalid marker when trying to see property information');
-      toast.warn('A map parcel must have a PID in order to view detailed information');
-    }
-  };
+  const mapRef = useRef<LeafletMap | null>(null);
 
-  const { setState, selectedInventoryProperty, selectedFeature } = useContext(MapStateContext);
+  const [activeFeatureLayer, setActiveFeatureLayer] = useState<L.GeoJSON>();
 
-  if (mapRef.current && !selectedInventoryProperty) {
-    const center = mapRef.current.getCenter();
-    lat = center.lat;
-    lng = center.lng;
+  // add geojson layer to the map
+  if (!!mapRef.current && !activeFeatureLayer) {
+    setActiveFeatureLayer(geoJSON().addTo(mapRef.current));
   }
 
-  const parcelLayerFeature = selectedFeature;
-  const { showLocationDetails } = useActiveFeatureLayer({
-    selectedProperty: selectedInventoryProperty,
-    layerPopup,
-    mapRef,
-    parcelLayerFeature,
-    setLayerPopup,
-  });
-
-  const lastZoom = useAppSelector(state => state.mapViewZoom) ?? zoomProp;
-  const [zoom, setZoom] = useState(lastZoom);
+  const handleMapClickEvent = (latlng: LatLng) => {
+    mapMachine.mapClick(latlng);
+  };
+
+  const [zoom, setZoom] = useState(DEFAULT_MAP_ZOOM);
+  const [isMapReady, setIsMapReady] = useState(false);
```

Comment:
it feels weird to have this outside the state machine, what do you think?

Reply:
This is necessary given the component might be ready but not the map. Having an effect with the isMapReady allows the map requests to be processed.
From the map fsm point of view, all map requests are async so it does not need to know if the map is ready.